python/lsst/daf/butler/core/config.py

```diff
@@ -70,6 +70,36 @@ def _doUpdate(d, u):
     return d
 
 
+def _checkNextItem(k, d, create, must_be_dict):
+    """See if k is in d and if it is return the new child."""
+    nextVal = None
+    isThere = False
+    if d is None:
+        # We have gone past the end of the hierarchy
+        pass
+    elif not must_be_dict and isinstance(d, collections.abc.Sequence):
+        # Check sequence first because for lists
+        # __contains__ checks whether value is found in list
+        # not whether the index exists in list. When we traverse
+        # the hierarchy we are interested in the index.
```

Comment:
I found "Check sequence first" confusing.  I think you mean checking if k is an index first before checking __contains__.